File: src/rsce/module/RSCEModule.java
package rsce.module;

import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;

import rsce.core.event.EventHandler;
import rsce.core.io.IOManager;
import rsce.entity.World;

import com.google.common.eventbus.AsyncEventBus;
import com.google.inject.AbstractModule;

/**
 * This serves as the configuration for all general aspects of RSCE.
 * 
 * @author Joe Pritzel
 * 
 */
public class RSCEModule extends AbstractModule {

	@Override
	protected void configure() {

		bind(World.class).toInstance(new World());
		
		bind(EventHandler.class).toInstance(
				new EventHandler(new AsyncEventBus(Executors
						.newSingleThreadExecutor()), Executors
						.newSingleThreadScheduledExecutor()));
		
		ScheduledExecutorService pool = Executors.newScheduledThreadPool(10);
		bind(IOManager.class).toInstance(
				new IOManager(new AsyncEventBus(pool), pool));

	}
}

File: src/main/scala/rsce/RSCEModule.scala
package rsce

import com.google.inject.AbstractModule
import rsce.entity.World
import rsce.entity.Player
import rsce.entity.PlayerProvider
import com.google.inject.name.Names
import com.joepritzel.feather.PSBrokerBuilder
import com.joepritzel.feather.strategy.publish.Sequential
import com.joepritzel.feather.PSBroker
import com.joepritzel.feather.strategy.publish.FewQuickListeners
import java.util.concurrent.Executors

/**
 * This serves as the configuration for all general aspects of RSCE.
 *
 * @author Joe Pritzel
 *
 */
class RSCEModule extends AbstractModule {
  override protected def configure {
    bind(classOf[World]).toInstance(new World)

    bind(classOf[Player]).toProvider(classOf[PlayerProvider])

    bind(classOf[PSBroker]).annotatedWith(Names.named("EventHandler")).toInstance(
      new PSBrokerBuilder().publishStrategy(new Sequential(true)).build)
      
    bind(classOf[PSBroker]).annotatedWith(Names.named("IOHandler")).toInstance(
      new PSBrokerBuilder().publishStrategy(new FewQuickListeners(
        Executors.newCachedThreadPool, true)).build)

  }
}